app/models/ticket.rb

```diff
@@ -21,21 +21,31 @@ def paid?(user)
     ticket_purchases.find_by(user: user, paid: true).present?
   end
 
-  def quantity_bought_by(user)
-    result = ticket_purchases.where(user_id: user.id).first
-    result ? result.quantity : 0
+  def quantity_bought_by(user, paid: false)
+    purchased_tickets = ticket_purchases.where(user_id: user.id, paid: paid)
```

Comment:
Actually, you can also much easily add up the quantities.

Look at the [sum method](http://api.rubyonrails.org/v4.2.5/classes/ActiveRecord/Calculations.html#method-i-sum).